integrations/extensions/starter-kits/google-custom-search/README.md

```diff
@@ -0,0 +1,153 @@
+- [Add the extension to your assistant](https://cloud.ibm.com/docs/watson-assistant?topic=watson-assistant-add-custom-extension) using the API key you obtained in the pre-requisites step above.
+- Create session variables to be used for storing the intermediate results.
+- Go to `Variables > Created by you` and add `query_text`, `cx`,`search_result`,`link`, `title`, `snippet`, `extension_result`,`exclude_terms`, `include_terms`,`exact_terms`,`date_restrict`.
+- Create variable `num_of_results` and set it to maximum number of results you want from the search api or set it `3` as default max number of results.
+- Follow the `basic` step below to get started with search and test it in the Preview chat.
+- Results can be filtered or scoped based on the given parameters with advanced filtered search. Once you are done with basic steps, follow the filtered search section for advanced setup using filter parameters in search api.
+  ![Variable](./assets/variables.png)<br>
+
+
+### Basic
+Once this starter kit is properly installed, you can issue a query to your bot and if there is no other action that you've configured that matched that query then it will generate search results for that query. 
+
+#### Search
+- Create new action and name it `Search`.
+- Click the fX button to add a variable and add new session variable `query_text` and select "Expression" type and then put `input.text` or `input.original_text` as the expression.  The former will employ spelling correction to fix any detected spelling errors before sending the query, which can be helpful but it can also be counterproductive if your documents include specialized terminology that is not in our dictionary (such as product names) so you can use `input.original_text` as the alternative in such cases.
+- Optional: In "Assistant says", put `Searching for: ${query_text}`.
+- In "And then", select "Use an extension", and select the search endpoint and set the `query` parameter to the `query_text` session variable and the `cx` parameter to `cx` session variable you set in pre-requisite step and `num` parameter to `num_of_results` session variable. 
+- In new step, Store the results returned by extension in variable `extension_result`.
+- In next steps, call the `Process Result` action to do further processing on the results stored in previous step.
+- If you are planning for advanced `Filtered Search` action. Then you need to add extra step asking user the type of filter they want to apply on the result. Based on the user response each filter will be selected at runtime.
+  ![Search Action](./assets/search_action.gif)<br>
+
+#### Process Result
+- Create new action and name it `Process Result`
+- change "without conditions" to "with conditions" and "check if `extension_result is not defined`  or `extension_result.body == null` or `extension_result.body.items == null` or `extension_result.body.items.size == 0`.
```

Comment:
I think this will trigger when you get no search results, which is not what you want because there is a separate message for no search results found in "Show Search Results".  So I think this should just check `extension_result.success==false`.  If you want to be very cautious, you could check  `extension_result.success==false` or `extension_result is not defined`

Reply:
Included the condition for "false" value in success field.